common/management/commands/create_test_data.py

```diff
@@ -0,0 +1,1775 @@
+            price_unit=random.choice(PriceUnit.values),
+            lowest_price=lowest,
+            lowest_price_net=lowest_net,
+            highest_price=highest,
+            highest_price_net=highest_net,
+            tax_percentage=tax_percentage,
+            status=pricing_status,
+            reservation_unit=reservation_unit,
+        )
+        pricings.append(pricing)
+
+    return ReservationUnitPricing.objects.bulk_create(pricings)
+
+
+@with_logs(
+    text_entering="Creating tax percentages...",
+    text_exiting="Tax percentages created!",
+)
+def _create_tax_percentages() -> list[TaxPercentage]:
+    tax_percentages: list[TaxPercentage] = []
+    percentages = (0, 10, 14, 24)
+    for percentage in percentages:
+        # Creation is done one-by-one since 'value' is a not defined as unique
+        # and 'bulk_create' doesn't support 'update_conflicts' without a unique constraint:
+        # https://docs.djangoproject.com/en/4.2/ref/models/querysets/#bulk-create
+        tax_percentage, _ = TaxPercentage.objects.get_or_create(value=percentage)
+        tax_percentages.append(tax_percentage)
+
+    return tax_percentages
+
+
+@with_logs(
+    text_entering="Creating reservations...",
+    text_exiting="Reservations created!",
+)
+def _create_reservations(
+    user: User,
+    reservation_units: list[ReservationUnit],
+    reservation_purposes: list[ReservationPurpose],
+    age_groups: list[AgeGroup],
+    cancel_reasons: list[ReservationCancelReason],
+    deny_reasons: list[ReservationDenyReason],
+    *,
+    number: int = 200,
+) -> list[Reservation]:
+    past_date_start = datetime(year=2021, month=1, day=1, tzinfo=timezone.utc)
+    half = number // 2
+
+    reservations: list[Reservation] = []
+    reservation_units_chosen: list[ReservationUnit] = []
+    for i in range(number):
+        if i == half:
+            past_date_start = past_date_start.replace(year=2027)
+
+        reservation_unit: ReservationUnit = random.choice(reservation_units)
+
+        persons = random.randint(
+            reservation_unit.min_persons, reservation_unit.max_persons
+        )
+        min_hours = math.ceil(
+            reservation_unit.min_reservation_duration.total_seconds() / 3600
+        )
+        max_hours = math.ceil(
+            reservation_unit.max_reservation_duration.total_seconds() / 3600
+        )
+
+        begin = past_date_start + timedelta(days=i)
+        begin = begin.replace(
+            hour=random.choice(range(8, 20 - max_hours)),
+            minute=random.choice([0, 15, 30, 45]),
+        )
+        end = begin + timedelta(hours=random.choice(range(min_hours, max_hours)))
+
+        state = STATE_CHOICES.CONFIRMED
+
+        pricing: ReservationUnitPricing = random.choice(
+            list(reservation_unit.pricings.all())
+        )
+        if pricing.highest_price != Decimal("0"):
+            state = random.choice(
+                [STATE_CHOICES.WAITING_FOR_PAYMENT, STATE_CHOICES.CONFIRMED]
```

Comment:
**Reservations**: might not need `state = WAITING_FOR_PAYMENT` ones, they would be deleted by batch job (and can't be paid before that since don't exist in Verkkokauppa).
But some `REQUIRES_HANDLING` could be useful? (Could be those with the non-zero price + `applying_for_free_of_charge` set to true and a `free_of_charge_reason` on those.)

Reply:
No confirmed_at then if REQUIRES_HANDLING -- or DENIED (which would have handled_at).

Reply:
Changed implemented.